index.js

```diff
@@ -1,77 +1,32 @@
 function timelydiff(timestamp, length = null) {
-
-    //Calculate the time difference between the current time and the timestamp, in seconds.
-    let timeDifference = (Date.now() - timestamp) / 1000;
-    let timeDirection = null;
-
-
-    //Determine whether the time difference is in the past or future (is negative)
-    //and sets the timeDirection variable accordingly.
-    //It also takes the absolute value of timeDifference in case it's negative.
-    if (timeDifference < 0) {
-        timeDifference = Math.abs(timeDifference);
-        timeDirection = 'future';
-    }
-    else {
-        timeDirection = 'past';
-    }
-
-
-    //This is an array of arrays, each containing three elements:
-    //a unit of time (short and default),
-    //the calculated value of that unit based on the timeDifference,
-    //and the maximum value that unit can have 
-    //(i.e. 60 seconds in a minute, 24 hours in a day, etc.).
-    const timeUnits = [
-        [{ short: 's', def: 'second' }, timeDifference, 60],
-        [{ short: 'm', def: 'minute' }, Math.round(timeDifference / 60), 60],
-        [{ short: 'h', def: 'hour' }, Math.round(timeDifference / 3600), 24],
-        [{ short: 'd', def: 'day' }, Math.round(timeDifference / 86400), 7],
-        [{ short: 'w', def: 'week' }, Math.round(timeDifference / 604800), 4],
-        [{ short: 'mo', def: 'month' }, Math.round(timeDifference / 2419200), 12],
-        [{ short: 'y', def: 'year' }, Math.round(timeDifference / 29030400), 10],
-        [{ short: 'dcd', def: 'decade' }, Math.round(timeDifference / 290304000), Infinity]
-    ];
-
-
-    //this helper function returns the time string with "in" or "ago" depending on the time direction.
-    function positionTime(timeString) {
-        if (timeDirection === 'future') {
-            return `in ${timeString}`;
-        } else {
-            return `${timeString}${length === 'shorter' ? '' : ' ago'}`;
-        }
+  const now = Date.now();
+  const timeDifference = Math.abs((now - timestamp) / 1000);
+  const timeDirection = now < timestamp ? "future" : "past";
+  const timeUnits = [
+    { short: "s", def: "second", max: 60 },
+    { short: "m", def: "minute", max: 60 },
+    { short: "h", def: "hour", max: 24 },
+    { short: "d", def: "day", max: 7 },
+    { short: "w", def: "week", max: 4 },
+    { short: "mo", def: "month", max: 12 },
+    { short: "y", def: "year", max: 10 },
+    { short: "dcd", def: "decade", max: Infinity },
+  ];
+
+  for (const unit of timeUnits) {
+    const { short, def, max } = unit;
+    const value = Math.round(timeDifference / (length === "short" ? 1 : max));
+    if (value < max) {
+      const pluralize = value > 1 ? "s" : "";
+      const timeString =
+        length === "shorter"
+          ? `${value}${short}`
+          : `${value} ${def}${pluralize}`;
+      return `${timeDirection === "future" ? "in " : ""}${timeString}${
+        length === "shorter" ? "" : " ago"
```

Comment:
if a date is in the future, and the length is `short`, this will say "in 2h ago", which isn't quite right